reproduction.py

```diff
@@ -0,0 +1,163 @@
+            .small-table { border-spacing: 0px; border-collapse: none; border-radius: 0px; margin-bottom: -1em; }
+            .small-table td { border: 0px none; width: 30px; height: 30px; }
+            .small-table tr { border: 0px none; width: 30px; height: 30px; }
+            .small-table .light { background: whitesmoke; }
+            .small-table .dark { background: sienna }
+        </style>
+    </head>
+    <body>
+        <table>
+            <tbody>
+                <tr>
+                    <td style="text-align: center">
+                        <h1> Original: </h1>
+                        <table class="chess-board">
+                            <tbody>
+                                %s
+                            </tbody>
+                        </table>
+                        <h2 style="margin-bottom: 0em">castling: %s </h2>
+                    </td>
+                    <td style="text-align: center">
+                        <h1> Reproducted: </h1>
+                        <table class="chess-board">
+                            <tbody>
+                                %s
+                            </tbody>
+                        </table>
+                        <h2 style="margin-bottom: 0em">castling: %s </h2>
+                    </td>
+                </tr>
+            </tbody>
+        </table>
+    </body>
+</html>'''
+
+small_table_temp = '''
+    <table class="small-table">
+    <tr>
+      <td class="{0}"><img src="{1[1]}" style="opacity:{1[0]}; height: 100%"></td>
+      <td class="{0}"><img src="{2[1]}" style="opacity:{2[0]}; height: 100%"></td>
+    </tr>
+    <tr>
+      <td class="{0}"><img src="{3[1]}" style="opacity:{3[0]}; height: 100%"></td>
+      <td class="{0}"><img src="{4[1]}" style="opacity:{4[0]}; height: 100%"></td>
+    </tr>
+    </table>'''
+
+
+def display_fen(fen):
+    fen = fen.split(' ')
+    board_fen = fen[0]
+    player = fen[1]
+    table = display_board(board_fen, player == 'w')
+    castling = fen[2].replace('K', '♔').replace('Q', '♕').replace('k', '♚').replace('q', '♛')
+    return table, castling
+
+
+def display_board(fen, player):
+    table = ''
+    fen = fen.split("/")
+    if not player:
+        fen = fen[::-1]
+    for y, i in enumerate(fen):
+        table += '<tr>\n'
+        x = 0
+        for j in i:
+            if j.isnumeric():
+                img = '<img src="{}">'.format(empty_img)
+                for k in range(int(j)):
+                    table += '\t<td class="{}">{}</td>\n'.format('dark' if (x + y) % 2 else 'light', img)
+                    x += 1
+            else:
+                img = '<img src="{}">'.format(pieces_name.format(pieces_path[j]))
+                table += '\t<td class="{}">{}</td>\n'.format('dark' if (x + y) % 2 else 'light', img)
+                x += 1
+        table += '</tr>\n'
+    return table
+
+
+def draw_reproduction(fen, model, treshold=0.1, cast_treshold=0.5):
+    inf = Inference(settings.DEVICE, model)
```

Comment:
I don't like the part that this function is named "draw_" but it does inference as well. How could you logically split that?